sys/filedesc.c

```diff
@@ -26,18 +26,51 @@ static void fd_mark_unused(fdtab_t *fdt, int fd) {
   bit_clear(fdt->fdt_map, fd);
 }
 
+/* Grows given file descriptor table to contain new_size file descriptors
+ * (up to MAXFILES)
+ */
+
+static void fd_growtable(fdtab_t *fdt, size_t new_size) {
```

Comment:
This function will only work correctly if the `fdt` mutex is locked. Please either lock it recursively, or simply `assert(mtx_locked(&fdt->fdt_mtx))`.